ert3/storage/_storage.py

```diff
@@ -356,13 +398,58 @@ def get_ensemble_record(
 ) -> ert3.data.EnsembleRecord:
     if experiment_name is None:
         experiment_name = f"{workspace}.{_ENSEMBLE_RECORDS}"
+    experiment = _get_experiment_by_name(experiment_name)
+    if experiment is None:
+        raise ert3.exceptions.NonExistantExperiment(
+            f"Cannot get {record_name} data, no experiment named: {experiment_name}"
+        )
 
-    return _get_numerical_data(workspace, experiment_name, record_name)
+    param_names = _get_experiment_parameters(workspace, experiment_name)
+    if record_name in param_names:
+        ensemble_records = [
+            _get_numerical_data(
+                workspace, experiment_name, f"{record_name}.{param_name}"
+            )
+            for param_name in param_names[record_name]
+        ]
+
+        # Combine records into the first ensemble record
+        combined_records: List[ert3.data.Record] = []
+        for record_idx, _ in enumerate(ensemble_records[0].records):
+            record0 = ensemble_records[0].records[record_idx]
+
+            if isinstance(record0.data, list):
+                ldata = [
+                    val
+                    for data in (
+                        ensemble_record.records[record_idx].data
+                        for ensemble_record in ensemble_records
+                    )
+                    if isinstance(data, list)
+                    for val in data
+                ]
+                combined_records.append(ert3.data.Record(data=ldata))
+            elif isinstance(record0.data, dict):
+                ddata = {
+                    key: val
+                    for data in (
+                        ensemble_record.records[record_idx].data
+                        for ensemble_record in ensemble_records
+                    )
+                    if isinstance(data, dict)
+                    for key, val in data.items()
+                }
+                combined_records.append(ert3.data.Record(data=ddata))
+
+        return ert3.data.EnsembleRecord(records=combined_records)
+    else:
+        return _get_numerical_data(workspace, experiment_name, record_name)
 
 
 def get_ensemble_record_names(
-    *, workspace: Path, experiment_name: Optional[str] = None
+    *, workspace: Path, experiment_name: Optional[str] = None, _flatten: bool = True
```

Comment:
Are we sure we want to expose this implementation detail only for the purpose of testing? 🤔